packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -71,54 +77,81 @@ async function runAndValidateBatch(
+                for (let i = 0; i < 10; i++) {
+                    testObject.root.set(i.toString(), i);
+                }
             });
-        const container = await loader.resolve({ url: containerUrl });
-        const testObject = await requestFluidObject<TestFluidObject>(container, "default");
-        // send batch
-        testObject.context.containerRuntime.orderSequentially(() => {
+            await new Promise((resolve) => setTimeout(resolve, 0));
+            // send non-batch
+            testObject.root.set("foo", "bar");
+            while (container.isDirty && !container.closed) {
+                await timeoutPromise((resolve, reject) => {
+                    container.once("saved", () => resolve());
+                    container.once("closed", (e) => reject(e));
+                },
+                {
+                    durationMs: timeout, // 60 * 60 * 1000,
+                });
+            }
+
+            assert.equal(container.closed, false, "container should not be closed");
+            assert.equal(container.isDirty, false, "container should not be dirty");
+
             for (let i = 0; i < 10; i++) {
-                testObject.root.set(i.toString(), i);
+                assert.equal(testObject.root.get(i.toString()), i, i.toString());
             }
-        });
-        // send non-batch
-        testObject.root.set("foo", "bar");
-        while (container.isDirty && !container.closed) {
-            await timeoutPromise((resolve, reject) => {
-                container.once("saved", () => resolve());
-                container.once("closed", (e) => reject(e));
-            },
-            {
-                durationMs: timeout, // 60 * 60 * 1000,
+            assert.equal(testObject.root.get("foo"), "bar", "validate after batch op");
+        }
+        // load a new container and validate there as well to ensure everything saved
+        {
+            const loader = provider.createLoader(
+                [[
+                    provider.defaultCodeDetails,
+                    provider.createFluidEntryPoint({ runtimeOptions: { summaryOptions: { disableSummaries: true } } }),
+                ]]);
+            const container = await loader.resolve({ url: containerUrl });
+            await timeoutAwait(waitContainerToCatchUp(container), {
+                durationMs: timeout,
             });
+            const testObject = await requestFluidObject<TestFluidObject>(container, "default");
+            for (let i = 0; i < 10; i++) {
+                assert.equal(testObject.root.get(i.toString()), i, `unexpected value after saved ${i.toString()}`);
+            }
+            assert.equal(testObject.root.get("foo"), "bar", "validate after save");
         }
-
-        for (let i = 0; i < 10; i++) {
-            assert.equal(testObject.root.get(i.toString()), i, i.toString());
+    } catch (e) {
+        if (isILoggingError(e)) {
+            return e;
```

Comment:
Don't you want to throw otherwise or something? Seems suspicious to swallow the error in that case.

Reply:
this isn't ready for check in. just trying to reproduce the issue, and see if we can gather more data

Reply:
@andre4i and @WayneFerrao and I were looking into some of these failures in Wayne's PR, Andrei had a theory that maybe sometimes the container would close before processing ops, or something like that, such that the expected errors wouldn't appear.

Reply:
Mark is correct. However, based on how error reporting works wrt to reading the logs (see https://github.com/microsoft/FluidFramework/blob/b915948f43ada3267ce1cef09f6e93abec2f2886/packages/test/test-utils/src/testObjectProvider.ts#L427), if the container closed with an error, we would definitely find out as the test would fail with the `Unexpected Errors in Logs` message and not try to match the declared errors at all.

The other theory is that the container closes without an error and there's no invocation of anything which would throw some sort of 'runtime is closed' assertion and then the silent close falls under the radar leading to the `Expected Events not found` error.